### ISSUE TITLE
Where to find anchor information for pretrained object detection model

### ISSUE THREAD

**Issue by Pixelfahnder** [google/mediapipe#2786]
Greetings,

> I'm refering to this model: https://github.com/google/mediapipe/blob/master/mediapipe/models/object_detection_saved_model/README.md

> I'm using **python** for testing the whole stuff

I'm fairly new to this whole topic and was happy to manage it to let the object detection model perform an inference in a sample code. After lots of research I understood, that the model input requires a 320x320 8 bit RGB image which I provided as an input to Tensorflow (I use the Tensorflow light variant of the network).

So after beeing able to perform a proper inference I came across, that I may have to parse the output data according to my needs and discovered, that there is an output of 2034 bounding boxes with 4 values as well as a buffer of 2034 by 91 values.
As I understood the 2024 outputs are the so called "anchor boxes" which were calculated randomly during the training process (or are they defined by the COCO dataset?). The 91 object classes seem to be the 91 elments per detection class of the anchor boxes.

What I'm missing is information about where the anchor boxes are located in the model? Is this information stored somehwere? Or do I have to use the mediapipe framework to calculate them on the fly? If yes, how would I do so?

You see I'm very new and very sorry to ask such a basic question. But I guess the hard stuff still will come as I read some stuff about "Non-max Suppression" and other stuff.

Basically what I want ot achieve is the proper way of showing bounding boxes with estimations about objects, so probably I need to do the whole post processing on my own and I struggle right now.

**Comment by ahmadyan** [comment]
Hi, 
For our SSD (Single Stage Calculators) inference (Mediapipe in an inference library, not a training library), [ssd_anchors_calculator.cc](https://github.com/google/mediapipe/blob/ecb5b5f44ab23ea620ef97a479407c699e424aa7/mediapipe/calculators/tflite/ssd_anchors_calculator.cc) computes the bounding boxes (an object may be detected multiple times) and then we perform NMS via [non_max_suppression_calculator](https://github.com/google/mediapipe/blob/ecb5b5f44ab23ea620ef97a479407c699e424aa7/mediapipe/calculators/util/non_max_suppression_calculator.cc). You can see how they work together in this [graph](https://github.com/google/mediapipe/blob/master/mediapipe/graphs/object_detection/object_detection_mobile_gpu.pbtxt).
